server/routes/rag.js

```diff
@@ -17,6 +17,14 @@ const getCollection = async () => {
   return collection;
 };
 
+const filterOutExclusions = (pairs, exclude) => {
+  return pairs.filter(({ question, answer }) => {
+    if (exclude.length === 0) return true;
```

Comment:
Rather than including this in the filter function and checking the length of `exclude` for every question, could we make this check once before filtering?